handler.go

```diff
@@ -384,7 +360,11 @@ func (s *handler) createError(err error) *respError {
 
 	var ed ErrorWithData
 	if errors.As(err, &ed) {
-		out.Data = ed.ErrorData()
+		if md, err := json.Marshal(ed.ErrorData()); err == nil {
+			out.Data = md
+		} else {
+			log.Warnf("Failed to marshal error data: %v", err)
+		}
```

Comment:
doing this on the server makes me think that maybe we should just automagically do it on the client too

Reply:
but ... this only works on `ErrorData()` so it doesn't really help us because we'd need a setter or to mess with `Data` and then it's not an interface anymore and requires a field with that name